app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -781,4 +782,11 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
     }
 
     override fun getWorkManagerConfiguration() = Builder().setMinimumLoggingLevel(INFO).build()
+
+    @OptIn(DelicateCoroutinesApi::class)
+    private fun downloadWallpapers() {
+        GlobalScope.launch {
```

Comment:
I like to switch the dispatcher closest to the code that actually needs to be on the different dispatcher. That way, calling code doesn't need to introspect the async code to discover it needs to change dispatchers. This is a [suggested practice](https://developer.android.com/kotlin/coroutines/coroutines-adv#main-safety), and is something that can be observed in [Room](https://medium.com/androiddevelopers/room-coroutines-422b786dc4c5) and in Retrofit I believe. 

That is, it's safe to call `suspend` functions defined in Room and Retrofit from any `Dispatcher`, because the libraries ensure they switch to `IO` before performing async tasks. I like to mirror that behavior.

Reply:
We have some previous discussion about using the `Dispatchers.Default` when was strictly needed  you could read more https://github.com/mozilla-mobile/fenix/issues/19574 @mcomella and @csadilek could provide more pointers as they were more involved on the ticket.